library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionListFragment.kt

```diff
@@ -79,20 +97,52 @@ internal class TransactionListFragment :
     }
 
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
-        return if (item.itemId == R.id.clear) {
-            AlertDialog.Builder(requireContext())
-                .setTitle(R.string.chucker_clear)
-                .setMessage(R.string.chucker_clear_http_confirmation)
-                .setPositiveButton(
-                    R.string.chucker_clear
-                ) { _, _ ->
-                    viewModel.clearTransactions()
+        return when (item.itemId) {
+            R.id.clear -> {
+                AlertDialog.Builder(requireContext())
+                    .setTitle(R.string.chucker_clear)
+                    .setMessage(R.string.chucker_clear_http_confirmation)
+                    .setPositiveButton(
+                        R.string.chucker_clear
+                    ) { _, _ ->
+                        viewModel.clearTransactions()
+                    }
+                    .setNegativeButton(R.string.chucker_cancel, null)
+                    .show()
+                true
+            }
+            R.id.export -> {
+                if (::transactions.isInitialized && transactions.isNotEmpty()) {
+                    AlertDialog.Builder(requireContext())
+                        .setTitle(R.string.chucker_export)
+                        .setMessage(R.string.chucker_export_http_confirmation)
+                        .setPositiveButton(R.string.chucker_export) { _, _ ->
+                            val file = cacheFileFactory.createExportFile()
+                            file.writeText(ShareUtils.getStringFromTransactions(transactions, requireContext()))
+
+                            val uri = FileProvider.getUriForFile(
+                                requireContext(),
+                                getString(R.string.chucker_provider_authority), file
+                            )
+
+                            val intent = Intent().apply {
```

Comment:
Please switch to the `ShareCompat.IntentBuilder` with chooser like it is done in `TransactionActivity` or `ThrowableActivity`.